### PR TITLE
feat: bundle grpc_health_probe in image

### LINKED ISSUE #386: Include grpc-health-probe execurtable as part of the Open Saves Dockerfile
**Is your feature request related to a problem? Please describe.**
The version of Kubernetes that is running in GKE does not offer a way to specify liveness or readiness probes using grpc endpoints that we can use in our production workload. The feature has been introduced, but it is currently in Beta for 1.24.

**Describe the solution you'd like**
The solution that has been suggested on a few blog relies on packaging an executable as part of the service using the gRPC health interface and specifying this in the k8s yaml deployment file.

Details at: https://github.com/grpc-ecosystem/grpc-health-probe

**Describe alternatives you've considered**
Replacing the gRPC health check interface by a http server, which is currently supported in k8s.

**Additional context**

### PR DESCRIPTION
**Pull request by hongalex**
**What type of PR is this?**
/kind feat

**What this PR does / Why we need it**:
Bundles the statically compiled [grpc_health_probe](https://github.com/grpc-ecosystem/grpc-health-probe) in the container image.

**Which issue(s) this PR fixes**:
Closes #386

**Special notes for your reviewer**:

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -62,6 +62,10 @@ COPY --from=amd64 /src/open-saves/build/collector /collector
 
 CMD ["/collector"]
 
+RUN GRPC_HEALTH_PROBE_VERSION=v0.4.12 && \
+    wget -qO/bin/grpc_health_probe https://github.com/grpc-ecosystem/grpc-health-probe/releases/download/${GRPC_HEALTH_PROBE_VERSION}/grpc_health_probe-linux-amd64 && \
+    chmod +x /bin/grpc_health_probe
+
 # Build the server image.
```
Comment (loranger2k) [inline Dockerfile:69]: The executable should probably be packaged in both the collector and the server.
For the collector, I would add this section at line 62 and line 78 in the original file, and keep the CMD as the last step.

**Review by loranger2k — CHANGES_REQUESTED** [review]
Add binaries for both collector and server.
I am not sure we even need it for the collector since it runs on a schedule and terminates when done.

File: Dockerfile (hunk 1 above)
Comment (hongalex) [reply Dockerfile:69]: Done

**Review by loranger2k — APPROVED** [review]
LGTM

**Pull request merged**